graph/path/shortest_test.go

```diff
@@ -262,43 +263,67 @@ func BenchmarkShortestAlts(b *testing.B) {
 }
 
 func BenchmarkAllShortest(b *testing.B) {
+	funcs := []struct {
+		name string
+		fn   func(g graph.Graph) AllShortest
+	}{
+		{
+			name: "DijkstraAllPaths",
+			fn:   DijkstraAllPaths,
+		},
+		{
+			name: "FloydWarshall",
+			fn: func(g graph.Graph) AllShortest {
+				p, _ := FloydWarshall(g)
+				return p
+			},
+		},
+	}
+
 	for _, bench := range shortestBenchmarks {
-		g := simple.NewDirectedGraph()
-		gen.SmallWorldsBB(g, bench.n, bench.d, bench.p, rand.New(rand.NewSource(bench.seed)))
-		p := DijkstraAllPaths(g)
+		for _, f := range funcs {
+			g := simple.NewDirectedGraph()
+			gen.SmallWorldsBB(g, bench.n, bench.d, bench.p, rand.New(rand.NewSource(bench.seed)))
+			p := f.fn(g)
 
-		// Find the widest path set.
-		var (
-			bestUid, bestVid int64
-			n                int
-		)
-		for uid := int64(0); uid < int64(bench.n); uid++ {
-			for vid := int64(0); vid < int64(bench.n); vid++ {
-				paths, _ := p.AllBetween(uid, vid)
-				if len(paths) > n {
-					n = len(paths)
-					bestUid = uid
-					bestVid = vid
+			// Find the widest path set.
+			var (
+				bestUid, bestVid int64
+				n                int
+			)
+			for uid := int64(0); uid < int64(bench.n); uid++ {
+				for vid := int64(0); vid < int64(bench.n); vid++ {
+					paths, _ := p.AllBetween(uid, vid)
+					if len(paths) > n {
+						n = len(paths)
+						bestUid = uid
+						bestVid = vid
+					}
 				}
 			}
-		}
 
-		b.Run(fmt.Sprintf("AllBetween_%d×%d|%v(%d)", bench.n, bench.d, bench.p, n), func(b *testing.B) {
-			for i := 0; i < b.N; i++ {
-				paths, _ := p.AllBetween(bestUid, bestVid)
-				if len(paths) != n {
-					b.Errorf("unexpected number of paths: got:%d want:%d", len(paths), n)
+			b.Run(fmt.Sprintf("%s_Between_%d×%d|%v(%d)", f.name, bench.n, bench.d, bench.p, n), func(b *testing.B) {
+				for i := 0; i < b.N; i++ {
+					_, _, _ = p.Between(bestUid, bestVid)
```

Comment:
Please don't dogsled. If you are concerned about elision, use a condition.

Reply:
Oh, great catch! Fixed.